packages/core/src/Site/index.js

```diff
@@ -280,6 +280,9 @@ class Site {
     const sourcePath = path.join(this.rootPath, config.pageSrc);
     const resultPath = path.join(this.outputPath, fsUtil.setExtension(config.pageSrc, '.html'));
     const codeTheme = this.siteConfig.style.codeTheme || 'dark';
+    // hide line numbers by default
+    const codeLineNumbers = this.siteConfig.style.codeLineNumbers !== undefined
```

Comment:
Could we centralise the initialisation of default values in `SiteConfig` directly?